Mathlib/Data/Int/NatAbs.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Jeremy Tan
+-/
+import Mathlib.Algebra.GroupWithZero.Hom
+import Mathlib.Algebra.GroupWithZero.Nat
+import Mathlib.Algebra.Ring.Int.Defs
+
+/-!
+# Lemmas about `Int.natAbs`
+
+This file contains some results on `Int.natAbs`, the absolute value of an integer as a
+natural number.
+
+## Main results
+
+* `Int.natAbsHom`: `Int.natAbs` bundled as a `MonoidWithZeroHom`.
+-/
+
+namespace Int
+
+/-- `Int.natAbs` as a bundled `MonoidWithZeroHom`. -/
+@[simps]
+def natAbsHom : ℤ →*₀ ℕ where
+  toFun := Int.natAbs
+  map_mul' := Int.natAbs_mul
+  map_one' := Int.natAbs_one
+  map_zero' := Int.natAbs_zero
+
+lemma natAbs_sub_nat_of_le {a b : ℕ} (h : b ≤ a) : Int.natAbs (↑a - ↑b) = a - b := by omega
```

Comment:
I'm seeing evidence that using `natCast` instead of just `nat` in these lemmas' names would be preferred:

https://github.com/leanprover-community/mathlib4/blob/0114eff24c4c8a5b4431dda340b46f39e1fdcb69/Mathlib/Algebra/Group/Basic.lean#L853-L855
https://github.com/leanprover-community/mathlib4/blob/0114eff24c4c8a5b4431dda340b46f39e1fdcb69/Mathlib/Algebra/Order/Round.lean#L97-L102

https://github.com/leanprover-community/mathlib4/pull/23563


So maybe it should be `natAbs_natCast_sub_natCast_of_le`?